CMakeLists.txt

```diff
@@ -60,49 +67,71 @@ else()
     set (CLANG_BIN ${CLANG_INSTALL_PREFIX}/bin/clang++)
 endif()
 
-execute_process(COMMAND ${CLANG_BIN} -print-resource-dir
-    RESULT_VARIABLE CLANG_RESOURCE_DIR_RESULT
-    OUTPUT_VARIABLE CLANG_RESOURCE_DIR
-    ERROR_VARIABLE  CLANG_RESOURCE_DIR_ERROR
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-if(CLANG_RESOURCE_DIR_RESULT)
-    message(FATAL_ERROR "Could not locate the resource dir for clang we are going to link against")
-    message(FATAL_ERROR ${CLANG_RESOURCE_DIR_ERROR})
-endif()
+# execute_process(COMMAND ${CLANG_BIN} -print-resource-dir
+#     RESULT_VARIABLE CLANG_RESOURCE_DIR_RESULT
+#     OUTPUT_VARIABLE CLANG_RESOURCE_DIR
+#     ERROR_VARIABLE  CLANG_RESOURCE_DIR_ERROR
+#     OUTPUT_STRIP_TRAILING_WHITESPACE)
+# 
+# if(CLANG_RESOURCE_DIR_RESULT)
+#     message(FATAL_ERROR "Could not locate the resource dir for clang we are going to link against")
+#     message(FATAL_ERROR ${CLANG_RESOURCE_DIR_ERROR})
+# endif()
 
 message(STATUS "Using the following resource dir for clang: ${CLANG_RESOURCE_DIR}")
 
 file(GLOB EMITTER_FILES CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/emitters/*.cpp)
```

Comment:
Beware when using globs in cmake: the underlying build system won't know when a file is added/removed so it won't refresh the list of sources unless you re-configure.

Reply:
A fair point. On the other hand, if a file is added, CMake won't care, will it? Because the file isn't explicitly mentioned here, and the CMakeLists.txt would require an explicit update. (I'm trying to understand the tradeoffs here before cutting over to explicitly listing all the sources.)

Reply:
You can see some discussion about that in the [cmake doc](https://cmake.org/cmake/help/latest/command/file.html#filesystem):
> Note We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.
or in this [StackOverflow answer](https://stackoverflow.com/a/1060061/20984) which nicely covers the pros and cons, or [this one](https://stackoverflow.com/a/32412044/20984).

Basically globbing can cause headaches by compiling unwanted files, or forgetting to compile some files after checkouting a different commit for instance.